sector-builder/src/scheduler.rs

```diff
@@ -83,13 +83,11 @@ impl Scheduler {
         // we should immediately restart sealing.
         //
         // For more information, see rust-fil-sector-builder/17.
-        let protos: Result<Vec<SealTaskPrototype>> = m
-            .get_staged_sector_filtered(Some(SealStatus::Sealing))
-            .into_iter()
-            .map(|meta| m.create_seal_task_proto(meta.sector_id))
-            .collect();
+        let protos = m.create_seal_task_protos(|x| {
+            x.seal_status == SealStatus::Sealing(Default::default())
```

Comment:
- I think having a method on `SealStatus` like `is_sealing` could make this nicer
- alternatively checking here with 

```rust
if let SealStatus::Sealing(_) = x.seal_status { true } else { false }
```

would avoid the need for adding a custom `partial_eq` impl

Reply:
Sounds good.